app/src/main/kotlin/compiler/ast/AstFactory.kt

```diff
@@ -9,7 +10,7 @@ import compiler.parser.grammar.ParserGrammar.Productions
 import compiler.parser.grammar.Symbol
 
 object AstFactory {
-    class AstCreationFailed : Throwable()
+    class AstCreationFailed : CompilationFailure()
```

Comment:
It would be cleaner to stick to one convention regarding naming: if we have `CompilationFailure` then it should be `AstCreationFailure`.`
Same with other exception classes from other files.

Reply:
Changed `CompilationFailure` to `CompilationFailed`.